ivadomed/preprocessing.py

```diff
@@ -0,0 +1,38 @@
+# Transformation to perform on image before network processing
+import nibabel as nib
+import numpy as np
+import ivadomed.utils as imed_utils
+
+
+def get_midslice_average(path_im, ind, slice_axis=0):
+    """
+    Extract an average 2D slice out of a 3D volume. This image is generated by
+    averaging the 7 slices in the middle of the volume
+    Args:
+        path_im (string): path to image
+        ind (int): index of the slice around which we will average
+        slice_axis (int): Slice axis according to RAS convention
+
+    Returns:
+        nifti: a single slice nifti object containing the average image in the image space.
+
+    """
+    image = nib.load(path_im)
+    image_can = nib.as_closest_canonical(image)
```

Comment:
is this necessary, given that the slice dimension is given as input? why do you need to do that?

Reply:
This is linked to the slice_axis (so if one is in RAS orientation the other one should be as well). I did that have a known orientation so that I don't have to check which axis is RL. It can be done. I just find it easier to always aligned in RAS, as most of the code does. 
When processing batch of images, it avoid finding the 'RL' axis [here](https://github.com/ivadomed/ivadomed/blob/48e9d9b2b17f60fff3c87d270652b402db7b8e78/ivadomed/scripts/prepare_dataset_vertebral_labeling.py#L88) which is what I was thinking about when coding this.

Reply:
i see. makes sense